roles/tests-integration/lib/sniffer.rs

```diff
@@ -59,7 +59,42 @@ pub struct Sniffer {
     messages_from_downstream: MessagesAggregator,
     messages_from_upstream: MessagesAggregator,
     check_on_drop: bool,
-    intercept_messages: Vec<InterceptMessage>,
+    action: Option<Action>,
+}
+
+/// Represents an action that [`Sniffer`] can take on intercepted messages.
+#[derive(Debug, Clone)]
+pub enum Action {
+    /// Blocks the message stream after encountering a specific message.
+    BlockFromMessage(BlockFromMessage),
+    /// Intercepts and modifies a message before forwarding it.
+    InterceptMessage(Box<InterceptMessage>),
+}
+
+/// Defines an action that blocks the message stream after detecting a specific message.
+#[derive(Debug, Clone)]
+pub struct BlockFromMessage {
+    direction: MessageDirection,
+    expected_message_type: MsgType,
+}
+
+impl BlockFromMessage {
+    /// Creates a new [`BlockFromMessage`] action.
+    ///
+    /// - `direction`: The direction of the message stream to block.
+    /// - `expected_message_type`: The type of message after which the stream should be blocked.
```

Comment:
IMO this is not necessary. while it is good to be verbose in documentation, we don't want to restate the obvious stuff.